resources/qml/WelcomePages/AddLocalPrinterScrollView.qml

```diff
@@ -0,0 +1,190 @@
+                    height: UM.Theme.getSize("standard_arrow").height
+                    sourceSize.width: width
+                    sourceSize.height: height
+                    color: UM.Theme.getColor("text")
+                    source: base.currentSection == section ? UM.Theme.getIcon("arrow_bottom") : UM.Theme.getIcon("arrow_right")
+                }
+
+                Label
+                {
+                    id: label
+                    anchors.left: arrow.right
+                    anchors.leftMargin: UM.Theme.getSize("default_margin").width
+                    verticalAlignment: Text.AlignVCenter
+                    text: button.text
+                    font.bold: true
+                    renderType: Text.NativeRendering
+                }
+            }
+
+            onClicked:
+            {
+                base.currentSection = section
+                base.updateCurrentItemUponSectionChange()
+            }
+        }
+    }
+
+    Component
+    {
+        id: machineButton
+
+        RadioButton
+        {
+            id: radioButton
+            anchors.left: parent.left
+            anchors.leftMargin: UM.Theme.getSize("standard_list_lineheight").width
+            anchors.right: parent.right
+            anchors.rightMargin: UM.Theme.getSize("default_margin").width
+            height: visible ? UM.Theme.getSize("standard_list_lineheight").height : 0
+
+            checked: ListView.view.currentIndex == index
+            text: name
+            font: UM.Theme.getFont("default")
+            visible: base.currentSection == section
+
+            background: Rectangle
+            {
+                anchors.fill: parent
+                color: "transparent"
+            }
+
+            indicator: Rectangle
+            {
+                implicitWidth: 16
+                implicitHeight: 16
+                anchors.verticalCenter: parent.verticalCenter
+                radius: width / 2
+                border.width: UM.Theme.getSize("default_lining").width
+                border.color: radioButton.hovered ? UM.Theme.getColor("small_button_text") : UM.Theme.getColor("small_button_text_hover")
+
+                Rectangle {
+                    width: parent.width / 2
+                    height: width
+                    anchors.centerIn: parent
+                    radius: width / 2
+                    color: radioButton.hovered ? UM.Theme.getColor("primary_button_hover") : UM.Theme.getColor("primary_button")
+                    visible: radioButton.checked
+                }
+            }
+
+            contentItem: Label
+            {
+                verticalAlignment: Text.AlignVCenter
+                leftPadding: radioButton.indicator.width + radioButton.spacing
+                text: radioButton.text
+                font: radioButton.font
+                renderType: Text.NativeRendering
+            }
+
+            onClicked:
+            {
+                ListView.view.currentIndex = index
```

Comment:
`ListView.view.currentIndex = index` works because this is bound to the `ListView` which uses this `RadioButton` Component as its delegate. Changing it to `machineList` also works, but then it won't be reusable at least.